src/app/handlers/welcome.handler.ts

```diff
@@ -0,0 +1,50 @@
+import { GuildMember, RichEmbed } from 'discord.js';
+import { IContainer, IHandler } from '../../common/types';
+import Constants from '../../common/constants';
+
+export class WelcomeHandler implements IHandler {
+  private _LION_URL: string = 'https://github.com/joey-colon/lion';
+
+  constructor(public container: IContainer) {}
+
+  public async execute(member: GuildMember): Promise<void> {
+    const hasAvatar = Boolean(member.user.avatar);
+    const embed = this._createEmbed(hasAvatar);
+    await member.send(embed).catch();
```

Comment:
if we hit the catch it means they have dms disabled. we should probably send it to a channel and maybe ping them (?)

Reply:
We would prob need a new channel with the message in it. Overwrite their perms for it if we hit the catch, ping. and take away overwrite after they acknowledge

Reply:
Might as well just have another channel only Unacknowledged can see

Reply:
What if they have/haven't been marked as unverified? You probably don't want that in there since not *everyone* is unverified

Reply:
maybe if they're unverified we should @ them in the verify channel.

Reply:
this would re-couple the welcome message with the roles, or at least require synchronisation.

Reply:
not necessarily, all new users are unacknowledged.
And we can check to see if they have a pfp to see if they will be unverified

Reply:
Also, ping in verify only if DMs are off?